blogs/scripts/taxonomy.js

```diff
@@ -1,83 +1,58 @@
 import { toClassName } from './lib-franklin.js';
 
 /**
- * Find the second-level list items and put them into a single array.
- * @returns {Array} An array of tag strings
+ * Retrieve tags from API.
+ * @param {String} [category] The category of tags to be returned (keysight|segmentation)
+ * @param {String} [lang] The language of the tags to be returned
+ * @returns {Array} An array of tag objects
  */
-async function getTags(ul) {
-  const tagList = ul.querySelectorAll('ul li ul li');
-  const tagArray = [...tagList];
-  const textArray = tagArray.map((li) => toClassName(li.textContent));
-  return textArray;
-}
-
-/**
- * Retrieve HTML from tags.docx.
- * @returns {Array} An array of tag strings
- */
-async function getFranklinTags() {
+async function getAEMTags(category = 'keysight', lang = 'en') {
   let resp;
   try {
-    // Fetch the HTML content of the webpage
-    resp = await fetch(`${origin}/blogs/tags.plain.html`);
+    resp = await fetch(`https://www.keysight.com/clientapi/search/aemtags/${lang}`);
     if (resp && resp.ok) {
-      const text = await resp.text();
-      const tempElement = document.createElement('div');
-      tempElement.innerHTML = text;
-      // Get the root <ul> element
-      const rootUlElement = tempElement.querySelector('ul');
-      // Create the JavaScript array from the nested <ul>
-      const result = await getTags(rootUlElement);
-      return result;
+      const json = await resp.json();
+      const tagObjs = json.hits.filter((entry) => entry.TAG_PATH.includes(`/${category}/`));
+      return tagObjs;
     }
   } catch (e) {
     // console.log('Error:', e);
   }
   return null;
 }
 
-/**
- * Validate an array of tag objects against a source.
- * @param {Array} tagObjArray The object array to validate.
- * @returns {Array} Array containing the valid tag objects.
- */
-export async function validateTagObjs(tagsObjArray) {
-  try {
-    const allowedTags = await getFranklinTags();
-    const result = [];
-    tagsObjArray.forEach((obj) => {
-      if (allowedTags.includes(toClassName(obj.tag))) {
-        result.push(obj);
-      }
-    });
-    return result;
-  } catch (e) {
-    // console.log('Error:', e);
-  }
-  return null;
-}
-
 /**
  * Validate an array of tag strings against a source.
  * @param {String} tagArray The string array to validate.
  * @returns {Array} Array containing two arrays. The first array being only the valid tags,
  * the second one being the tags that are invalid
  */
-export async function validateTags(tagsArray) {
+export default async function validateTags(tagsArray) {
```

Comment:
it may make sense to add 2 optional params on this function for category and language.